WordPress/gutenberg
# Form Input: Don't use `flex-direction: row-reverse` for checkbox field

Follow up #63081

## What?

This PR dynamically changes the markup to remove `flex-direction: row-reverse;` from the Form Input block.

## Why?

In #63081, a stylelint rule was added to disallow flex-direction reverse values.

In the Form Input block, `flex-direction: row-reverse;` is applied to place text after the checkbox element. Currently, it is allowed with a disable comment, but ideally, this rule should not be used.

## How?

If the `type` attribute is `checkbox` or `radio`, place the label after the input element. This means that the HTML will be changed to the following:

```html

<div class="wp-block-form-input">
	<label class="wp-block-form-input__label">
		<span class="wp-block-form-input__label-content">Label</span>
		<input class="wp-block-form-input__input" type="checkbox" name="label" aria-required="false"/>
	</label>
</div>


<div class="wp-block-form-input">
	<label class="wp-block-form-input__label">
		<input class="wp-block-form-input__input" type="checkbox" name="label" aria-required="false"/>
		<span class="wp-block-form-input__label-content">Label</span>
	</label>
</div>
```

Also added block deprecation since we are updating the `save` function.

Note: Updating the HTML for the Form Input block was once attempted in #63081, but was ultimately removed due to the need for block deprecation. The code in this PR is based on [this commit](https://github.com/WordPress/gutenberg/pull/63081/commits/db35911bb0445c5de8291304032f34609d72014e).

## Testing Instructions

- Switch to the trunk branch.
- Through the code editor, insert the following HTML and save the post. This is a single Form Input block, and the input element is a checkbox:
  ```html
  
  <div class="wp-block-form-input">
  	<label class="wp-block-form-input__label">
  		<span class="wp-block-form-input__label-content">Label</span>
  		<input class="wp-block-form-input__input" type="checkbox" name="label" aria-required="false"/>
  	</label>
  </div>
  
  ```
- On the front end, make sure the checkbox is ordered first, then the label.
- Switch to the `form-input/remove-row-reverse` branch.
- Reload the browser on the front end.
- The order of the checkbox and the label should remain unchanged.
- Open the post editor, update the checkbox label, and save the post. The markup will be updated as follows:
  ```html
  
  <div class="wp-block-form-input">
  	<label class="wp-block-form-input__label">
  		<input class="wp-block-form-input__input" type="checkbox" name="label" aria-required="false"/>
  		<span class="wp-block-form-input__label-content">Label</span>
  	</label>
  </div>
  
  ```
- Reload the browser on the front end.
- The order of the checkbox and the label should remain unchanged.
